### DISCUSSION TITLE
Pyarmor group license

### DISCUSSION THREAD

**Discussion by Kathir2023** [dashingsoft/pyarmor#1322] [Q&A]
Hi Team,

We bought Group license and registered with below steps and received pyarmor-regfile-xxx.zip.

tried to register with .zip file based on steps provided in the email but getting error.
# version
pyarmor -v
Pyarmor 8.2.7 (trial), 000000, non-profits

License Type    : pyarmor-trial
License No.     : pyarmor-vax-000000
License To      : 
License Product : non-profits

BCC Mode        : No
RFT Mode        : No

Notes
* Can't obfuscate big script and mix str
* 
pyarmor register 
Pyarmor 8.0+ has only 3 commands: gen, reg, cfg
Please replace `pyarmor` with `pyarmor-7` to run old commands
kathirvel@kathirvels-MacBook-Air privagpt % pyarmor reg ./pyarmor-regfile-6033.zip
INFO     Python 3.11.4
INFO     Pyarmor 8.2.7 (trial), 000000, non-profits
INFO     Platform darwin.aarch64
INFO     register "./pyarmor-regfile-6033.zip"
ERROR    wrong usage for group license, please check `pyarmor reg` in Man page

## register command
pyarmor reg -g 1      
INFO     Python 3.11.4
INFO     Pyarmor 8.2.7 (trial), 000000, non-profits
INFO     Platform darwin.aarch64
INFO     generating device file ".pyarmor/group/pyarmor-group-device.1"
ERROR    dlopen(/opt/homebrew/lib/python3.11/site-packages/pyarmor/cli/core/pytransform3.so, 0x0002): Library not loaded: /Library/Frameworks/Python.framework/Versions/3.11/Python
  Referenced from: <130C5EB5-D506-36CD-B49A-E022E5293C40> /opt/homebrew/lib/python3.11/site-packages/pyarmor/cli/core/pytransform3.so
  Reason: tried: '/Library/Frameworks/Python.framework/Versions/3.11/Python' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Library/Frameworks/Python.framework/Versions/3.11/Python' (no such file), '/Library/Frameworks/Python.framework/Versions/3.11/Python' (no such file), '/System/Library/Frameworks/Python.framework/Versions/3.11/Python' (no such file, not in dyld cache)

**Comment by jondy** [comment]
How about apply these commands?
```
% name=pytransform3
% deplib=/Library/Frameworks/Python.framework/Versions/3.11/Python
% ver=3.11
% target=/opt/homebrew/lib/python3.11/site-packages/pyarmor/cli/core/pytransform3.so
% install_name_tool -id $name.so \
            -change $deplib @rpath/lib/libpython$ver.dylib \
            -add_rpath @executable_path/.. \
            -add_rpath @loader_path/.. \
            -add_rpath /System/Library/Frameworks/Python.framework/Versions/$ver \
            -add_rpath /Library/Frameworks/Python.framework/Versions/$ver \
            $target
        codesign -f -s - --all-architectures --timestamp $target
```

**Reply by Kathir2023** [reply]
for me worked in ubuntu VM but still issue with mac 

error:
pyarmor reg -g 2                                     
INFO     Python 3.11.4
INFO     Pyarmor 8.2.7 (trial), 000000, non-profits
INFO     Platform darwin.aarch64
INFO     generating device file ".pyarmor/group/pyarmor-group-device.2"
ERROR    dlopen(/opt/homebrew/lib/python3.11/site-packages/pyarmor/cli/core/pytransform3.so, 0x0002): Library not loaded: /Library/Frameworks/Python.framework/Versions/3.11/Python
  Referenced from: <130C5EB5-D506-36CD-B49A-E022E5293C40> /opt/homebrew/lib/python3.11/site-packages/pyarmor/cli/core/pytransform3.so
  Reason: tried: '/Library/Frameworks/Python.framework/Versions/3.11/Python' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Library/Frameworks/Python.framework/Versions/3.11/Python' (no such file), '/Library/Frameworks/Python.framework/Versions/3.11/Python' (no such file), '/System/Library/Frameworks/Python.framework/Versions/3.11/Python' (no such file, not in dyld cache)

**Comment by jondy** [comment]
If Python is not system Python in Darwin, refer to https://pyarmor.readthedocs.io/en/stable/reference/errors.html#outer-errors
to adapt Python path by install_name_tool

**Reply by Kathir2023** [reply]
sure,let me check. i got output 
MacBook-Air ~ % which python3
/opt/homebrew/bin/python3
